include/aspect/material_model/utilities.h

```diff
@@ -259,6 +266,88 @@ namespace aspect
                           const bool interpol,
                           const MPI_Comm &comm);
         };
+
+        /**
+        * This class reads in entropy-pressure material table and looks up for material
+        * proporties when entropy and pressure are given
+        */
+        class EntropyReader
+        {
+          public:
+
+            /**
+             * Read the table into material_lookup
+             */
+            void
+            initialize(const MPI_Comm comm,
+                       const std::string data_directory,
+                       const std::string material_file_name);
+
+            /**
+             * Reads the specific heat for a given entropy and pressure from material_lookup
+             */
+            double
+            specific_heat(const double entropy,
+                          const double pressure) const;
+
+            /**
+             * Reads the density for a given entropy and pressure from material_lookup
+             */
+            double
+            density(const double entropy,
+                    const double pressure) const;
+
+            /**
+             * Reads the thermal_expansivity for a given entropy and pressure
+             * from material_lookup
+             */
+            double
+            thermal_expansivity(const double entropy,
+                                const double pressure) const;
+
+            /**
+             * Reads the compressibility for a given entropy and pressure from material_lookup
+             */
+            double
+            compressibility(const double entropy,
+                            const double pressure) const;
+
+            /**
+             * Reads the temperature for a given entropy and pressure from material_lookup
+             */
+            double
+            temperature(const double entropy,
+                        const double pressure) const;
+
+            /**
+             * Reads the seismic p wave velocity for a given entropy and pressure
+             * from material_lookup
+             */
+            double
+            seismic_vp(const double entropy,
+                       const double pressure) const;
+
+            /**
+             * Reads the seismic s wave velocity for a given entropy and pressure
+             * from material_lookup
+             */
+            double
+            seismic_vs(const double entropy,
+                       const double pressure) const;
+
+            /**
+             * Gets density gradient for a given entropy and pressure
+             */
+            Tensor<1, 2>
+            calc_density_gradient(const double entropy,
+                                  const double pressure) const;
```

Comment:
The name of this function deviates from the pattern you established before. By prefixing it with `calc_` it suggests that there is a more complicated operation to perform. However the operation is not much more difficult (it simply calls `get_gradient` instead of `get_value` of the StructuredDataLookup class). Therefore I would suggest you just use the same naming scheme, which would mean  `density_gradient` would be sufficient as name.